cvat-ui/src/utils/opencv-wrapper/clahe.ts

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2021-2022 Intel Corporation
+// Copyright (C) 2023 CVAT.ai Corporation
+//
+// SPDX-License-Identifier: MIT
+
+import { BaseImageFilter, ImageProcessing } from 'utils/image-processing';
+
+export interface CLAHE extends ImageProcessing {
+    processImage: (src: ImageData, frameNumber: number) => ImageData;
+}
+
+export default class CLAHEImplementation extends BaseImageFilter {
+    private cv:any;
+    // same defaults as opencv
+    private clipLimit: number = 40.0;
+    private tileGridSize: { rows: number, columns: number } = { rows: 8, columns: 8 };
+
+    constructor(cv:any) {
+        super();
+        this.cv = cv;
+    }
+
+    public configure(options: {
+        clipLimit?: number,
+        tileGridSize?: { width: number, height: number }
+    }): void {
+        if (options.clipLimit !== undefined) {
+            this.clipLimit = options.clipLimit;
+        }
+        if (options.tileGridSize !== undefined) {
+            this.tileGridSize = options.tileGridSize;
+        }
+    }
+
+    public processImage(src:ImageData, frameNumber: number) : ImageData {
```

Comment:
Much of the code here is duplicated from `histogram-equalization.ts`. Could we create a base class to centralize the shared logic, particularly for the image processing functionality?